test/spmd/test_xla_distributed_checkpoint.py

```diff
@@ -208,6 +208,26 @@ def test_state_dict_separation(self):
           planner.unsharded_state_dict,
           ['fc1.weight', 'fc1.bias', 'fc2.weight', 'fc2.bias'])
 
+  def test_save_state_dict_with_cpu_shards(self):
```

Comment:
Let's add two more tests - one to verify the number of WriteItem generated, and one to verify the resolve_data call returns the local shards.